contracts/core/custodial-ownership-token/v1/ERC2258.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.6.11;
+
+import "lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import "lib/openzeppelin-contracts/contracts/math/SafeMath.sol";
+
+import "./interfaces/IERC2258.sol";
+
+/// @title ERC2258 implements the basic level functionality for a token capable of custodial ownership.
+contract ERC2258 is IERC2258, ERC20 {
+    using SafeMath       for uint256;
+
+    mapping(address => mapping(address => uint256)) public override custodyAllowance;
+    mapping(address => uint256)                     public override totalCustodyAllowance;
+
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) public { }
+
+    function increaseCustodyAllowance(address custodian, uint256 amount) external override {
+        uint256 oldAllowance      = custodyAllowance[msg.sender][custodian];
+        uint256 newAllowance      = oldAllowance.add(amount);
+        uint256 newTotalAllowance = totalCustodyAllowance[msg.sender].add(amount);
+
+        require(newTotalAllowance <= balanceOf(msg.sender), "ERC2258:INSUF_BALANCE");
```

Comment:
```js
require(custodian != address(0),                    "ERC2258:INVALID_CUSTODIAN");
require(amount    != uint256(0),                    "ERC2258:INVALID_AMT");
require(newTotalAllowance <= balanceOf(msg.sender), "ERC2258:INSUF_BALANCE");

Reply:
Both of the first checks are UX, and I purposely left them out of the `ERC2258`, as per our "principles".